### PR TITLE
show image volume when kubectl describe a pod with image volume

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

**pod.yaml**
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: pod
  annotations:
    example.com/foo: v
spec:
  containers:
    - name: test
      image: registry.k8s.io/e2e-test-images/echoserver:2.3
      env:
        - name: k
          valueFrom:
            fieldRef:
              fieldPath: metadata.annotations['example.com/foo']
      volumeMounts:
        - name: volume
          mountPath: /volume
  volumes:
    - name: volume
      image:
        reference: quay.io/crio/artifact:v1
        pullPolicy: IfNotPresent
```

```
(base) ➜  kubernetes git:(kubectl-describe-image-volume) kubectl describe po
Name:             pod
Namespace:        default
Priority:         0
Service Account:  default
Node:             crio-control-plane/172.18.0.2
Start Time:       Thu, 15 Aug 2024 17:07:19 +0800
Labels:           <none>
Annotations:      example.com/foo: v
Status:           Running
IP:               10.244.0.5
IPs:
  IP:  10.244.0.5
Containers:
  test:
    Container ID:   cri-o://074b1d107a561ef22013dcca58c7a78419923d34035f81535f83751425a99707
    Image:          registry.k8s.io/e2e-test-images/echoserver:2.3
    Image ID:       registry.k8s.io/e2e-test-images/echoserver@sha256:60bd75eb661054404b49f745694714b0c3298c698d8f08d21180189581e6a0f2
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Thu, 15 Aug 2024 17:07:37 +0800
    Ready:          True
    Restart Count:  0
    Environment:
      k:   (v1:metadata.annotations['example.com/foo'])
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-7lxfd (ro)
      /volume from volume (rw)
Conditions:
  Type                        Status
  PodReadyToStartContainers   True
  Initialized                 True
  Ready                       True
  ContainersReady             True
  PodScheduled                True
Volumes:
  volume:
  <unknown>
  kube-api-access-7lxfd:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age                 From               Message
  ----    ------     ----                ----               -------
  Normal  Scheduled  55m                 default-scheduler  Successfully assigned default/pod to crio-control-plane
  Normal  Pulling    55m                 kubelet            Pulling image "quay.io/crio/artifact:v1"
  Normal  Pulled     55m                 kubelet            Successfully pulled image "quay.io/crio/artifact:v1" in 9.747s (9.747s including waiting). Image size: 11235 bytes.
  Normal  Pulling    55m                 kubelet            Pulling image "registry.k8s.io/e2e-test-images/echoserver:2.3"
  Normal  Pulled     55m                 kubelet            Successfully pulled image "registry.k8s.io/e2e-test-images/echoserver:2.3" in 7.892s (7.892s including waiting). Image size: 24838677 bytes.
  Normal  Created    55m                 kubelet            Created container test
  Normal  Started    55m                 kubelet            Started container test
  Normal  Pulled     11s (x46 over 55m)  kubelet            Container image "quay.io/crio/artifact:v1" already present on machine
```

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/enhancements/issues/4639

#### Does this PR introduce a user-facing change?

```release-note
Fix kubectl doesn't print image volume when kubectl describe a pod with that volume
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/4639
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/describe/describe.go
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/describe/describe.go:1487]: In my opinion, this is a very long text and hurting the usability especially on narrow screens.

Comment (carlory) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go:1487]: How about it?
```
"Type:\tImage (a container image or OCI artifact)"
```

Comment (ardaguclu) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go:1487]: Sounds better

Comment (carlory) [reply staging/src/k8s.io/kubectl/pkg/describe/describe.go:1487]: I copy it from the first sentence of type definition like other volume type.
```
  image	<ImageVolumeSource>
    image represents an OCI object (a container image or artifact) pulled and
    mounted on the kubelet's host machine. The volume is resolved at pod startup
    depending on which PullPolicy value is provided:
...
  iscsi	<ISCSIVolumeSource>
    iscsi represents an ISCSI Disk resource that is attached to a kubelet's host
    machine and then exposed to the pod. More info:
    https://examples.k8s.io/volumes/iscsi/README.md

  name	<string> -required-
    name of the volume. Must be a DNS_LABEL and unique within the pod. More
    info:
    https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names

  nfs	<NFSVolumeSource>
    nfs represents an NFS mount on the host that shares a pod's lifetime More
    info: https://kubernetes.io/docs/concepts/storage/volumes#nfs

  persistentVolumeClaim	<PersistentVolumeClaimVolumeSource>
    persistentVolumeClaimVolumeSource represents a reference to a
    PersistentVolumeClaim in the same namespace. More info:
    https://kubernetes.io/docs/concepts/storage/persistent-volumes#persistentvolumeclaims

  photonPersistentDisk	<PhotonPersistentDiskVolumeSource>
    photonPersistentDisk represents a PhotonController persistent disk attached
    and mounted on kubelets host machine

  portworxVolume	<PortworxVolumeSource>
    portworxVolume represents a portworx volume attached and mounted on kubelets
    host machine

  projected	<ProjectedVolumeSource>
    projected items for all in one resources secrets, configmaps, and downward
    API

  quobyte	<QuobyteVolumeSource>
    quobyte represents a Quobyte mount on the host that shares a pod's lifetime

  rbd	<RBDVolumeSource>
    rbd represents a Rados Block Device mount on the host that shares a pod's
    lifetime. More info: https://examples.k8s.io/volumes/rbd/README.md

  scaleIO	<ScaleIOVolumeSource>
    scaleIO represents a ScaleIO persistent volume attached and mounted on
    Kubernetes nodes.

  secret	<SecretVolumeSource>
    secret represents a secret that should populate this volume. More info:
    https://kubernetes.io/docs/concepts/storage/volumes#secret

  storageos	<StorageOSVolumeSource>
    storageOS represents a StorageOS volume attached and mounted on Kubernetes
    nodes.

  vsphereVolume	<VsphereVirtualDiskVolumeSource>
    vsphereVolume represents a vSphere volume attached and mounted on kubelets
    host machine
```

**Comment by carlory** [comment]
@ardaguclu updated.

**Comment by ardaguclu** [comment]
Thank you. This looks good to me. Before the approval, I'd defer to sig-node or sig-storage for lgtm.

**Review by saschagrunert — APPROVED** [review]
Thank you!

/lgtm

**Comment by saschagrunert** [comment]
Is this something we need to backport? Is it an enhancement or more something I've missed during the implementation?

**Comment by carlory** [comment]
> Is this something we need to backport? 

No, the field is gated by a feature flag, and the feature flag is not enabled in the stable branch. so we don't need to backport this. @saschagrunert 

Refer to the [What Kind of PRs are Good for Cherry Picks](https://github.com/kubernetes/community/blob/80687eee3e061c9f25545e5554cd9bcc3f2603d9/contributors/devel/sig-release/cherry-picks.md#what-kind-of-prs-are-good-for-cherry-picks)

> Is it an enhancement or more something I've missed during the implementation?

I only encounter this issue when I try to use the feature.

**Pull request merged**
